### ISSUE TITLE
Party IDs and Historical Names

### ISSUE THREAD

**Issue by BobBorges** [swerik-project/riksdagen-persons#25]

**Comment by BobBorges** [comment]
@salgo60 Magnus, do you  know who to ping on wikidata to get some action under a proposal in organizations?

https://www.wikidata.org/wiki/Wikidata:Property_proposal/SWERIK_Party_ID

**Comment by salgo60** [comment]
The sad answer its a WD is a very very small community but do as you do now pinging me
* maybe add something on sv:Wikipedia [Wikipedia:Wikidatafrågor](https://sv.wikipedia.org/wiki/Wikipedia:Wikidatafr%C3%A5gor)
* there are also some Telegram groups for wikidata but I am not active there anymore I am more a Mapillary person now....

I will add some comments on [Wikidata:Property proposal/SWERIK Party ID ](https://www.wikidata.org/wiki/Wikidata:Property_proposal/SWERIK_Party_ID)

**Comment by salgo60** [comment]
Did a [small video](https://youtu.be/pyZXfEE5eqs)

<img width="912" alt="image" src="https://github.com/user-attachments/assets/30ab0590-cd65-4f26-b493-df3f02660d4e">

**Comment by salgo60** [comment]
* my vote is **oppose**
  * I believe you’re facing [technical debt](https://en.wikipedia.org/wiki/Technical_debt) if external domains are required to establish new properties specifically for your platform as you develop and expand your information model.
  
<img width="1252" alt="image" src="https://github.com/user-attachments/assets/d1d9c334-fc9e-4f77-ac45-d8adf8e5cc9a">

**Comment by salgo60** [comment]
>>  replace Q-IDs with SWERIK party IDs

Suggestion: still have the **Wikidata Q number in your data on the party object** and have versioning an all party objects so we can see how it change
 

If someone would like to match your parties and have "same as" Wikidata then they will have easier to match your parties 

>> also document "vilde"
* [#139 the "mess" I have found in different sources at different time periods](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/139)

**Comment by salgo60** [comment]
<img width="1249" alt="image" src="https://github.com/user-attachments/assets/ae938aa6-6b37-47ac-8851-5731b6a4f037">

**Comment by salgo60** [comment]
* what to do in WD [Property_talk:P12192#Change_of_this_external_identifier](https://www.wikidata.org/wiki/Property_talk:P12192#Change_of_this_external_identifier)

<img width="1241" alt="image" src="https://github.com/user-attachments/assets/c07c7ea5-83a2-4d72-91ae-67b5de493fac">

**Comment by salgo60** [comment]
Changed status to withdrawn

<img width="1084" alt="image" src="https://github.com/user-attachments/assets/ef75b043-cb9d-4757-8f48-778a577aacab">

**Comment by salgo60** [comment]
>> We want to represent the data as it is -- if party was called X at a certain time, it is an analytical decision to call X Y now because we think X and Y are the same. For this reason we created a successor tree and those who prefer to use Y for everything can do that.

I believe that significant changes, like two parties forming an alliance and beginning to work closely together, offer much more insight than simply tracking a name change. Such alliances can reshape political influence, strategies, and voter dynamics, making them more critical for analysis than a rebranding. In my view, these collaborative shifts reveal deeper and more meaningful transformations within the political landscape.

**Comment by salgo60** [comment]
me WD banned so I cant help you any more see [27#issuecomment-2456263922](https://github.com/swerik-project/riksdagen-persons/issues/27#issuecomment-2456263922)

**Comment by salgo60** [comment]
## OT we can now filter people with books in the Swedish Literature bank on partners

* did the below search yesterday when I tried to [explain the ParlaMint vision for Litterature bank people #59](https://github.com/Litteraturbanken/littb-frontend/issues/59#issuecomment-2697504991) 
   * I guess could learn something from you and your experience with TEI

### Swedish PM in Swedish Litteraturbanken in a Timeline with filter on parties

* Swedish PM with a book [in Swedish Litterature bank approx 160  on a timeline](https://js.histropedia.com/apps/query-timeline/?q=SELECT%20distinct%20?person%20?personLabel%20(SAMPLE(?SPA)%20AS%20?SPA)%20%20(sample(?bild)%20AS%20?bild)%20%0A?birth%20?death%20?partyLabel%20?Littbank%20WHERE%20%7B%0A%0A%20%20VALUES%20?member%20%7B%0A%20%20%20%20wd:Q33071890%20%0A%20%20%20%20wd:Q81531912%20%0A%20%20%20%20wd:Q82697153%20%0A%20%20%20%20wd:Q10655178%20%0A%20%20%7D%0A%20%20?person%20wdt:P39%20?member;%0A%20%20%20%20wdt:P1343%20?source.%0A%20%20?person%20p:P1343%20?pTva.%0A%20%20OPTIONAL%7B?person%20wdt:P102%20?party%7D%0A%20%20OPTIONAL%7B?person%20wdt:P569%20?birth%7D%0A%20%20OPTIONAL%7B?person%20wdt:P570%20?death%7D%0A%20%20?pTva%20ps:P1343%20wd:Q110346241.%0A%20%20?pTva%20prov:wasDerivedFrom%20%5B%20pr:P4819%20?SPAid%20%5D.%0A%0A%20%20OPTIONAL%20%7B?person%20wdt:P18%20?bild%7D.%0A%20%20%0A%20%20?person%20wdt:P5101%20?Litt.%20%0A%20%20BIND(URI(CONCAT(%22https://litteraturbanken.se/f%25C3%25B6rfattare/%22,?Litt))%20AS%20?Littbank)%0A%20%0A%20%20BIND(URI(CONCAT(%22https://portrattarkiv.se/details/%22,?SPAid))%20AS%20?SPA)%0A%0A%20%20SERVICE%20wikibase:label%20%7B%20bd:serviceParam%20wikibase:language%20%22sv,en%22.%20%7D%0A%7D%20GROUP%20BY%20%20?person%20?personLabel%20%20?death%20?birth%20?partyLabel%20?Littbank%0Aorder%20by%20?partyLabel&md=true&g=article&l=Littbank&t=personLabel&s=birth&e=death&i=bild&d=0&c=partyLabel&f=partyLabel&v=t) / [lista](https://w.wiki/D4Jd)

<img width="1449" alt="Image" src="https://github.com/user-attachments/assets/2f005d09-0131-400c-9df4-fca90e48280e" />

**Comment by BobBorges** [comment]
@salgo60 

>  do you have linked your party ids to Wikidata Qnumbers

yes -- for the moment it's only in the swerik data, though I would like to add it to wikidata

>  what is your party modell? like follow, followed by, inception, chairman

Any name change of the party name (whether the party renames itself, or it dissolves and a new party forms in it's place) constitutes a new ID, we have inception and dissolution dates and a list of successors (which can be length 0, 1 or more).

> have you linked your party data to any other database with political parties than Wikidata

This is the intention, but it's not done yet. When the proposal for a swerik party ID property was cancelled on WD, I lost my momentum. I guess we need to update the swerik person ID to include persons and party organizations. I haven't looked into how that works.

**Comment by salgo60** [comment]
Thanks I guess parties overtime can be more controversial than Swedish PMs

Would be cool if we in Wikidata could mark party items as "SWEriks approved".... I did a test to mark Swedish PMs party belonging as need more care...

@Lottabrorsson any ideas if we could make books related to Swedish PMs more easy to access like the timeline above? Feels like the Litteraturebank has good momentum maybe start to suggest what books they should add?!??

**Comment by BobBorges** [comment]
@salgo60 How would you suggest reopening the issue of changing the SWERIK person id to a more general SWERIK ID  in Wikidata, to that they could be applied to persons *and* political parties, and maybe expanded down the line to other scopes (for instance, we're now doing the same with the Finnish ståndriksdagen... those people and parties will need IDs too).

**Comment by salgo60** [comment]
A try to get Litteraturbanken involved 

![image](https://github.com/user-attachments/assets/c7b09a86-4ce8-40e8-a083-887647b1b2f9)

**Comment by salgo60** [comment]
> [@salgo60](https://github.com/salgo60) How would you suggest reopening the issue of changing the SWERIK person id to a more general SWERIK ID in Wikidata, to that they could be applied to persons _and_ political parties, and maybe expanded down the line to other scopes (for instance, we're now doing the same with the Finnish ståndriksdagen... those people and parties will need IDs too).

I am leaving Sunday for a week skiing in Andorra we can have a meeting after that....

I guess make some lists of people / parties and if you have the Wikidata is

I/you can then upload them to Wikidata everything is easier if we have external identifiers in common see test below

I normally use Open refine to match with Wikidata we have also tools like [Mix'n'match](https://meta.wikimedia.org/wiki/Mix%27n%27match) for matching external databases with WIkidata....

The problem I feel with parties is it some chaos at least with Swedish Communist related parties in Wikidata and I dont have the skills... same with Finnish ståndriksdagen I have no skills.

----
Small check what I see in Wikipedia

* WD [Q738695](https://www.wikidata.org/wiki/Q738695) [Prime Minister of Finland](https://en.wikipedia.org/wiki/Prime_Minister_of_Finland)
   *.[List of Prime ministers](https://en.wikipedia.org/wiki/List_of_prime_ministers_of_Finland)
  
#### Finnish Primeministers external identifiers  wdt:P31 wd:Q738695

* [list Finnish External identifiers for wd:Q738695](https://w.wiki/DLQZ) / [List all external identifiers](https://w.wiki/DLQq)

<img width="1453" alt="Image" src="https://github.com/user-attachments/assets/7986cc53-8093-4e65-88c8-50efd213c8e8" />

* Finnish Ministers database ID (new) [Property:P12910](https://www.wikidata.org/wiki/Property:P12910)
* Finnish Ministers database ID (archived) [Property:P2182](https://www.wikidata.org/wiki/Property:P2182)
* [discussion about Property:P12910](https://www.wikidata.org/wiki/Wikidata:Property_proposal/Finnish_Ministers_database_ID_(new))

_[finländsk minister-ID (arkiverad) (P2182)](https://www.wikidata.org/wiki/Property:P2182) doesn't work anymore since the IDs of the service have changed from numbers to strings, and thus it needs a replacement. All the statements from P2182 can be copied to the new property (except for URL format and regex). Full list of the new IDs can be found [here](https://fi.wikipedia.org/wiki/K%C3%A4ytt%C3%A4j%C3%A4:Anr/mk). –[Samoasambia](https://www.wikidata.org/wiki/User:Samoasambia) [✎](https://www.wikidata.org/wiki/User_talk:Samoasambia) [20:03, 15 July 2024 (UTC)](https://www.wikidata.org/wiki/Wikidata:Property_proposal/Finnish_Ministers_database_ID_(new)#c-Samoasambia-20240715200300-Motivering)_



  


Magnus Sälgö
0735152802
salgo60@msn.com

**Comment by salgo60** [comment]
### member of the Parliament of Finland 
* [Q17592486](https://www.wikidata.org/wiki/Q17592486) member of the Parliament of Finland
* [list Finnish External identifiers for wd:Q17592486](https://w.wiki/DLWG) / list [all external identifiers](https://w.wiki/DLWJ)

<img width="1439" alt="Image" src="https://github.com/user-attachments/assets/d95bf8d1-a71b-49f5-ba35-495611215d44" />

<img width="1437" alt="Image" src="https://github.com/user-attachments/assets/8048fcd9-589f-4ac5-a632-411c41490f39" />

**Comment by salgo60** [comment]
* check Estates' Diet 1905 
   * [sv:Wikipedia Lantdagen 1905](https://sv.wikipedia.org/wiki/Lantdagen_1905) looks like a lot of [red links](https://en.wikipedia.org/wiki/Wikipedia:Red_link) in Wikipedia and I can see no good Wikidata for 
   * [fi:WIkipedia Säätyvaltiopäivät 1905](https://fi.wikipedia.org/wiki/S%C3%A4%C3%A4tyvaltiop%C3%A4iv%C3%A4t_1905)
   * Wikidata feels like good

### Part of Parliament of Finland Q643412
wdt:P361 wd:Q643412

**Comment by salgo60** [comment]
## [P2181](https://www.wikidata.org/wiki/Property:P2181) Finnish MP ID

**Comment by ljo** [comment]
Highlight the definition and make clear that means in relation to other sources. And especially what to do when biographies and verified attributions are clashing.

**Comment by BobBorges** [comment]
@BobBorges formulate scope and provenance of party affiliation as a decision --> add caveat to readme

**Comment by salgo60** [comment]
@BobBorges I'm back in Sweden. Feel free to reach out if you think I can add value: 0046705937579.

You should note that I have no knowledge of Finnish Parliament members...

**Comment by BobBorges** [comment]
I think the only thing I need a little push with is how to edit the scope of the swerik ID property. If I add instance political party to the subject type constraint, does that do the trick? If I'm on the right track, a little follwo up question: I don't see any kind of logic in the statements -- will wikidata understand that an item with the ID can be an instance of a human OR a political party (not human AND political party)? If it's more complicated than this, I will call you :D



![Image](https://github.com/user-attachments/assets/abfc2d02-a6d1-4fc1-88bc-45e14b4c3f2e)

**Comment by salgo60** [comment]
@BobBorges 

>> If I add instance political party to the subject type constraint, does that do the trick?

Yes that will fix the property constraints see [Help:Property_constraints_portal ](https://www.wikidata.org/wiki/Help:Property_constraints_portal) 

This is a "soft" check i.e. you can add also other things but you dont get the warning and its not on the [Wikidata:Database reports/Constraint violations/P12192](https://www.wikidata.org/wiki/Wikidata:Database_reports/Constraint_violations/P12192)

Example of warning when adding SWERIKS to [Q105112](https://www.wikidata.org/wiki/Q105112) - Swedish Social Democratic Party ([history Q105112](https://www.wikidata.org/w/index.php?title=Q105112&action=history))

<img width="925" alt="Image" src="https://github.com/user-attachments/assets/6c0d5793-e33a-41a7-8250-df23f9398ed1" />

----
Also good to change [P31 instance of ](https://www.wikidata.org/wiki/Property:P12192)

<img width="1068" alt="Image" src="https://github.com/user-attachments/assets/67d8a74c-2d17-4b99-a06a-32a3a1acf3af" />

**Comment by Lottabrorsson** [comment]
> Thanks I guess parties overtime can be more controversial than Swedish PMs
> 
> Would be cool if we in Wikidata could mark party items as "SWEriks approved".... I did a test to mark Swedish PMs party belonging as need more care...
> 
> [@Lottabrorsson](https://github.com/Lottabrorsson) any ideas if we could make books related to Swedish PMs more easy to access like the timeline above? Feels like the Litteraturebank has good momentum maybe start to suggest what books they should add?!??

Thanks for the question. No, it's not something I've/we've thought about.

**Comment by salgo60** [comment]
Thanks @Lottabrorsson

I mentioned regarding '[politisk vilde #139](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/139),' various sources classify it differently. I've compiled some metadata on the topic, and I believe it would be valuable to include metadata that captures a broader range of researchers' perspectives and opinions from the time it occurred with **sources that states the facts**.

<img width="840" alt="Image" src="https://github.com/user-attachments/assets/8591f21b-5105-4a22-b81d-2f2d795f33e0" />
